packages/terra-date-picker/src/react-datepicker/day.jsx

```diff
@@ -206,8 +213,8 @@ class Day extends React.Component {
     const dayClassName = (this.props.dayClassName ? this.props.dayClassName(date) : undefined)
     return ['react-datepicker-day', dayClassName, 'react-datepicker-day--' + getDayOfWeekCode(this.props.day), {
       'react-datepicker-day--disabled': this.isDisabled(),
-      'react-datepicker-day--selected': this.isSameDay(this.props.selected),
-      'react-datepicker-day--selected-border': this.isSameDay(this.props.preSelection) && (document.activeElement.tagName === 'DIV' || document.activeElement === document.querySelector('[class*="previous"]')) && this.props.isCalendarKeyboardFocused,
+      'react-datepicker-day--selected': this.isSameDay(this.props.selected) && !isDayDisabled(this.props.selected, this.props),
+      'react-datepicker-day--selected-border': this.isSameDay(this.props.preSelection) && !isDayDisabled(this.props.selected, this.props) && (document.activeElement.tagName === 'DIV' || document.activeElement === document.querySelector('[class*="previous"]')) && this.props.isCalendarKeyboardFocused,
```

Comment:
When navigating to a different day using the keyboard, the preselected day doesn't lose the focus style when focus is shifted to another element.

![Jul-07-2020 14-04-44](https://user-images.githubusercontent.com/22618655/86830399-43c8a000-c05b-11ea-812b-d359afb5005d.gif)

Reply:
This was happening due to the preventDefault which was added in mouseDown event of day component. preventDefault was blocking render that happens on mouseDown hence causing selected-border to be left on the pre-selected day.  Removed preventDefault() and clearing the pre-selected day on mouse down for inline calendar, since it will make previous-selected / selected day to get focus border on holding mouse on day.
fix commit : 3977a24f43c51003cd112c871c73af6770a6af97

![date-picker-focus](https://user-images.githubusercontent.com/12869809/87386576-707e3980-c5be-11ea-9477-c761051febe9.gif)